static/js/pages/instances/actions/FreezeInstanceBtn.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { FC, useState } from "react";
+import { LxdInstance } from "types/instance";
+import { useQueryClient } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { NotificationHelper } from "types/notification";
+import ConfirmationButton from "components/ConfirmationButton";
+import { Tooltip } from "@canonical/react-components";
+import { freezeInstance, unfreezeInstance } from "api/instances";
+
+interface Props {
+  instance: LxdInstance;
+  notify: NotificationHelper;
+  hasCaption?: boolean;
+  isDense?: boolean;
+}
+
+const FreezeInstanceBtn: FC<Props> = ({
+  instance,
+  notify,
+  hasCaption = true,
+  isDense = true,
+}) => {
+  const [isLoading, setLoading] = useState(false);
+  const queryClient = useQueryClient();
+
+  const handleFreeze = () => {
+    setLoading(true);
+    freezeInstance(instance)
+      .then(() => {
+        setLoading(false);
+        void queryClient.invalidateQueries({
+          queryKey: [queryKeys.instances],
+        });
+        notify.success(`Instance ${instance.name} frozen.`);
+      })
+      .catch((e) => {
+        setLoading(false);
+        notify.failure("Error on instance freeze.", e);
+      });
+  };
+
+  const handleUnfreeze = () => {
+    setLoading(true);
+    unfreezeInstance(instance)
+      .then(() => {
+        setLoading(false);
+        void queryClient.invalidateQueries({
+          queryKey: [queryKeys.instances],
+        });
+        notify.success(`Instance ${instance.name} is now running.`);
+      })
+      .catch((e) => {
+        setLoading(false);
+        notify.failure("Error on instance unfreeze.", e);
+      });
+  };
+
+  const isDisabled =
+    isLoading || instance.status === "Stopped" || instance.status === "Error";
+
+  return (
+    <Tooltip
+      message={isDisabled ? "Instance must be running" : undefined}
```

Comment:
I am a bit confused, this would show up when loading as well.

Reply:
Fixed @ [12afd4e](https://github.com/canonical/lxd-ui/commit/12afd4ee74f4b5eec68e0e678f3e45d68e280243)